test/distributed/fsdp/test_fsdp_state_dict.py

```diff
@@ -157,6 +165,40 @@ def _get_simple_model(self, *fsdp_args, checkpoint_wrap=False, **fsdp_kwargs):
         model = FSDP(lin, *fsdp_args, **fsdp_kwargs)
         return model
 
+    def _get_multibuffer_nested_model(
+        self, *fsdp_args, wrap=True, checkpoint_wrap=False, **fsdp_kwargs
+    ):
+        full_p = torch.float32
+        lin_mp = fsdp_kwargs.pop("mixed_precision", None)
+        bn_mp = (
+            MixedPrecision(param_dtype=full_p, reduce_dtype=full_p, buffer_dtype=full_p)
+            if lin_mp
+            else None
+        )
+        if wrap:
+            lin1 = nn.Linear(10, 10, bias=False).cuda()
+            bn1 = nn.BatchNorm1d(10).cuda()
+            lin2 = nn.Linear(10, 10, bias=False).cuda()
+            if checkpoint_wrap:
+                lin1 = checkpoint_wrapper(lin1)
+                bn1 = checkpoint_wrapper(bn1)
+                lin2 = checkpoint_wrapper(lin2)
+            seq = nn.Sequential(
+                FSDP(lin1, mixed_precision=lin_mp, *fsdp_args, **fsdp_kwargs),
+                FSDP(bn1, mixed_precision=bn_mp, *fsdp_args, **fsdp_kwargs),
+                lin2,
+            )
+            if checkpoint_wrap:
+                seq = checkpoint_wrapper(seq)
+            model = FSDP(seq, *fsdp_args, **fsdp_kwargs)
+        else:
+            model = nn.Sequential(
+                nn.Linear(10, 10, bias=False).cuda(),
+                nn.BatchNorm1d(10).cuda(),
+                nn.Linear(10, 10, bias=False).cuda(),
+            )
+        return model
```

Comment:
just for my understanding, where do the multi buffers come from? Is it the multiple BN units?

Reply:
The "multi-buffers" is just referencing the multiple buffers in the BN module (e.g. 'running_var', 'running_mean' etc.). I thought it was worth emphasizing/testing particularly because the bug here does not rear its ugly head for modules with a single buffer.